File: app/src/main/java/com/chernybro/loftcoin/LoftApp.java
package com.chernybro.loftcoin;

import android.app.Application;
import android.os.StrictMode;

import androidx.viewbinding.BuildConfig;

public class LoftApp extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        if (BuildConfig.DEBUG) {
            StrictMode.enableDefaults();
        }
    }
}

File: app/src/main/java/com/chernybro/loftcoin/LoftApp.kt
package com.chernybro.loftcoin

import android.app.Application
import android.os.StrictMode
import com.chernybro.loftcoin.utils.logging.DebugTree
import timber.log.Timber

class LoftApp : Application() {

    private lateinit var component: BaseComponent

    override fun onCreate() {
        super.onCreate()
        if (BuildConfig.DEBUG) {
            StrictMode.enableDefaults()
            Timber.plant(DebugTree())
        }
        component = DaggerAppComponent.builder()
            .application(this)
            .build()
    }

    fun getComponent(): BaseComponent {
        return component
    }

}